modules/planning/common/reference_line_info.cc

```diff
@@ -790,6 +790,20 @@ void ReferenceLineInfo::ExportEngageAdvice(EngageAdvice* engage_advice) const {
   if (!prev_advice->has_advice()) {
     prev_advice->set_advice(EngageAdvice::DISALLOW_ENGAGE);
   }
+
+  // when started from offlane
+  if (prev_advice->advice() == EngageAdvice::READY_TO_ENGAGE) {
+    if (vehicle_state_.driving_mode() !=
+        Chassis::DrivingMode::Chassis_DrivingMode_COMPLETE_AUTO_DRIVE) {
+      prev_advice->set_advice(EngageAdvice::READY_TO_ENGAGE);
+    } else {
+      prev_advice->set_advice(EngageAdvice::KEEP_ENGAGED);
+    }
+    prev_advice->clear_reason();
+    engage_advice->CopyFrom(*prev_advice);
+    return;
```

Comment:
This would change the engage advice logic for on road planning, and all engage in situations that we previously did not intend to engage, thus cause potential danger.

Reply:
Updated. Thank you!